framework/core/src/Extension/ExtensionManager.php

```diff
@@ -302,7 +304,17 @@ public function getEnabledExtensions(): array
      */
     public function extend(Container $container): void
     {
-        foreach ($this->getEnabledExtensions() as $extension) {
+        $extensions = $this->getEnabledExtensions();
+
+        if ($this->maintenance->isSafeMode()) {
```

Comment:
Should this run before dependency resolution? Seems possible that this could be used to enable a dependent of something disabled.

Reply:
nice catch 🤔

Reply:
![image](https://github.com/flarum/framework/assets/20267363/81f1d128-a3ed-48c9-a04b-a93e5e9f5679)
![image](https://github.com/flarum/framework/assets/20267363/d789ece9-9f2c-4d8f-993d-f8b7f2a4fdff)

Reply:
Good feature, but I think some help text / an icon tooltip / something is needed to explain why the disabled buttons are disabled.